lib/krb5/pac.c

```diff
@@ -383,7 +383,7 @@ krb5_pac_add_buffer(krb5_context context, krb5_pac p,
     size_t len, offset, header_end, old_end;
     uint32_t i;
 
-    assert(data->length > 0 && data->data != NULL);
+    assert(data->length == 0 || (data->length > 0 && data->data != NULL));
```

Comment:
Our requirement post 190263bb7a56fc775b50a6cd0dc91820d2b2e5eb is that any valid krb5_data must not have a NULL data pointer.   A valid krb5_data that represents a length of zero must have a non-NULL data pointer.   If both the length is zero and the data pointer is NULL, then then krb5_data is unallocated (it might have already been freed.).